### PR TITLE
Upload UI

### PR DESCRIPTION
**Pull request by RhythmAgg**
Upload Ui upgraded

### CODE REVIEW

**Comment by RhythmAgg** [comment]
https://github.com/UmangSharma5/DFS_ReactViewer/assets/92436585/f45cdef2-740a-46a0-9351-bbd9264fb524

The UI changes are reflected here

File: client/src/components/Viewer/Viewer.css [hunk 1]
```diff
@@ -8,18 +8,24 @@
...
+  width: 100%;
 }
 
+/* @media only screen and (min-width: 768px) {
+  .form-container {
+    min-width: 300px;
+  }
+  
+} */
+
```
Comment (shaantanu314) [inline client/src/components/Viewer/Viewer.css:28]: remove

File: client/src/components/Viewer/Viewer.js [hunk 2]
```diff
@@ -37,169 +40,188 @@ function Viewer() {
...
   }
 
+  // function handleChange(e) {
+  //   const file = e.target.files[0];
+  //   setCurrentFile(prevValue => ({
+  //     ...prevValue,
+  //     name: file,
+  //   }));
+  // }
+
```
Comment (shaantanu314) [inline client/src/components/Viewer/Viewer.js:50]: remove

File: client/src/components/Viewer/Viewer.js [hunk 3]
```diff
@@ -7,11 +7,13 @@ import { config } from '../Config/config';
...
 import { io } from 'socket.io-client';
 import StatusInfo from '../statusInfo';
+import Modal from 'react-bootstrap/Modal';
+import Button from 'react-bootstrap/Button';
 
 function Viewer() {
   const [currentFile, setCurrentFile] = useState({
     count: 0,
-    name: '',
+    names: '',
```
Comment (shaantanu314) [inline client/src/components/Viewer/Viewer.js:16]: this `name` variable is confusing. This is file and not filename. 
Please use `files` as variable name

File: client/src/components/Viewer/Viewer.js [hunk 4]
```diff
@@ -37,169 +40,188 @@ function Viewer() {
...
-    }
+      if (res !== undefined && res.data.isUploaded === 1) {
+        toast.warn('Image Already Exists');
+        setDisplayProgressBar(false);
+        return;
+      }
+
+      const socket = io(config.SOCKET_URL, { path: '/hv/socket' });
+      socket.on('connect', () => {
+        // console.error('Connected:', socket.connected); // Should be true
```
Comment (shaantanu314) [inline client/src/components/Viewer/Viewer.js:80]: remove

File: client/src/components/Viewer/Viewer.js [hunk 5]
```diff
@@ -37,169 +40,188 @@ function Viewer() {
...
-    socket.on('AddedUser', async () => {
-      const formData = new FormData();
-      setDisplayProgressBar(true);
-      formData.append('file', currentFile.name);
-      let bucketURL = config.BASE_URL + '/objects/' + shortEmail;
+      socket.on('AddedUser', async () => {
+        const formData = new FormData();
+        setDisplayProgressBar(true);
+        formData.append('file', file);
+        let bucketURL = config.BASE_URL + '/objects/' + shortEmail;
```
Comment (shaantanu314) [inline client/src/components/Viewer/Viewer.js:139]: what is this socket message used for?

File: client/src/components/Viewer/Viewer.js [hunk 6]
```diff
@@ -37,169 +40,188 @@ function Viewer() {
...
+                (progressEvent.loaded / progressEvent.total) * 100,
+              );
+              setProgressValue(percentCompleted);
+            },
+          });
+          if (response.status === 200) {
+            // toast.info('Upload is in Progress....Please check after some time');
+          } else {
+            toast.error('Error in Uploading File');
+          }
```
Comment (shaantanu314) [inline client/src/components/Viewer/Viewer.js:163]: if statement is redundant.

better

`if(response.status !== 200){...}`

File: client/src/components/Viewer/Viewer.js [hunk 7]
```diff
@@ -37,169 +40,188 @@ function Viewer() {
...
-          ...prevValue,
-          name: response.data.filename,
-          format: response.data.format,
-          count: prevValue.count + 1,
-        }));
-      } catch (error) {
-        console.error(error);
-      }
-      inProgressUpload.current -= 1;
+        inProgressUpload.current -= 1;
```
Comment (shaantanu314) [inline client/src/components/Viewer/Viewer.js:179]: use of this variable?

File: client/src/components/Viewer/Viewer.js [hunk 8]
```diff
@@ -21,6 +23,7 @@ function Viewer() {
   const [uploadPercentage, setUploadPercentage] = useState({});
   const [isConnected, setIsConnected] = useState(false);
   const inProgressUpload = useRef(0);
+  const [show, setShow] = useState(false);
```
Comment (shaantanu314) [inline client/src/components/Viewer/Viewer.js:26]: use proper names

`const [showUploadModal, setShowUploadModal] = ueState(false)`

File: client/src/components/statusInfo.js [hunk 9]
```diff
@@ -6,6 +6,7 @@ import BootstrapProgressBar from './bootstrapProgressBar';
 
 const StatusInfo = props => {
   const [showThumbnails, setShowThumbnails] = useState(false);
+  console.log(props.uploadPercentage);
```
Comment (shaantanu314) [inline client/src/components/statusInfo.js:9]: remove

File: client/src/components/Viewer/Viewer.css (hunk 1 above)
Comment (RhythmAgg) [reply client/src/components/Viewer/Viewer.css:28]: resolved

File: client/src/components/Viewer/Viewer.js (hunk 2 above)
Comment (RhythmAgg) [reply client/src/components/Viewer/Viewer.js:50]: resolved

File: client/src/components/Viewer/Viewer.js (hunk 3 above)
Comment (RhythmAgg) [reply client/src/components/Viewer/Viewer.js:16]: resolved

Comment (RhythmAgg) [reply client/src/components/Viewer/Viewer.js:16]: resolved

File: client/src/components/Viewer/Viewer.js (hunk 4 above)
Comment (RhythmAgg) [reply client/src/components/Viewer/Viewer.js:80]: resolved

File: client/src/components/Viewer/Viewer.js (hunk 5 above)
Comment (RhythmAgg) [reply client/src/components/Viewer/Viewer.js:139]: this mssg is sent from the server to the client to tell that the user has been added and thus the API request can now be sent. Thus it establishes a flow like socket connect -> user added -> API request

File: client/src/components/Viewer/Viewer.js (hunk 6 above)
Comment (RhythmAgg) [reply client/src/components/Viewer/Viewer.js:163]: resolved

File: client/src/components/Viewer/Viewer.js (hunk 7 above)
Comment (RhythmAgg) [reply client/src/components/Viewer/Viewer.js:179]: was reduntant, now removed

File: client/src/components/Viewer/Viewer.js (hunk 8 above)
Comment (RhythmAgg) [reply client/src/components/Viewer/Viewer.js:26]: resolved

File: client/src/components/statusInfo.js (hunk 9 above)
Comment (RhythmAgg) [reply client/src/components/statusInfo.js:9]: resolved

**Pull request merged**